src/client/src/components/List.jsx

```diff
@@ -1,30 +1,19 @@
 import React from "react";
 import PropTypes from "prop-types";
-import Item from "./Item"
+import Item from "./Item";
+import ItemConnector from "./ItemConnector";
+import useFilterByQuery from "../hooks/useFilterByQuery"; 
 
-
-const List = ({ allItems, itemToDelete, itemToEdit }) => {
+const List = ({ items }) => {
+  const filter = useFilterByQuery();
+  
   return (
     <div>
-      {allItems.map((item) => {
-        return (
-          <Item
-            key={item.itemId}
-            item={item}
-            itemToDelete={itemToDelete}
-            itemToEdit={itemToEdit}
-          />
-        );
+      {filter(items).map((item) => {
+        return <ItemConnector key={item.itemId} item={item} />;
```

Comment:
passing the entire "item" is a bad practice. you should pass only specific props that this component needs.
Or you can pass the ID only, and in the ItemConnector you can get the item from the state and pass the relevant props to the component